src/index.ts

```diff
@@ -145,7 +162,18 @@ export class UnleashClient extends TinyEmitter {
             try {
                 const context = this.context;
                 const urlWithQuery = new URL(this.url.toString());
-                Object.keys(context).forEach((key) => urlWithQuery.searchParams.append(key, context[key]));
+                // Add context information to url search params. If the properties
+                // object is included in the context, flatten it into the search params
+                // e.g. /?...&property.param1=param1Value&property.param2=param2Value
+                Object.entries(context).forEach((contextEntry) => {
```

Comment:
I switched from Object.keys to Object.entries because I feel that it's cleaner. Less object accesses. Also saves some non-null assertions I believe. Can definitely switch back if necessary!